src_new/util.js

```diff
@@ -1487,4 +1496,49 @@ exports.getDevicePlatform = function(){
 		refThis.logError("Unable to get device platform" , ex);
 	}
 	return deviceType;
-}
+};
+
+exports.updateAdUnitsWithEids = function(adUnits){
+	if(refThis.isArray(adUnits)){
+		adUnits.forEach(function(adUnit){
+			adUnit.bids.forEach(function(bid){
+				refThis.updateUserIds(bid);
+			});
+		});
+	} else if(!refThis.isEmptyObject(adUnits)){
+		adUnits.bids.forEach(function(bid){
+			refThis.updateUserIds(bid);
+		});
+	}
+};
+
+exports.updateUserIds = function(bid){
+	// refThis.idsAppendedToAdUnits =true;
+	if(refThis.isUndefined(bid.userId)){
+		bid["userId"] = refThis.getUserIds();
+	}
+	else if(bid.userId){
+		/* istanbul ignore next */
+		bid.userId = Object.assign(bid.userId, refThis.getUserIds());
+	}
+	if(refThis.isUndefined(bid.userIdAsEids)){
+		bid["userIdAsEids"] = refThis.getUserIdsAsEids();
+	}
+	else if(refThis.isArray(bid.userIdAsEids)){
+		var idsPresent = new Set();
+		var ids = refThis.getUserIdsAsEids().concat(bid.userIdAsEids);
```

Comment:
adUnits are not expected to have the  userIdAsEids already set as it is added thru code in pbjs.
We can remove this code of checking and trying to find union.

Reply:
I've added a hook into requestBids as well where it might be possible where pbjs has already appended the userIds. Is it good to assume that we won't get ids ?

Reply:
i am saying only about userIdAsEids, it is good to assume that it will not be present as it is not suggested to do so anywhere :)

Reply:
@pm-harshad-mane  as discussed since we get userIds and userIdAsEids in beforeRequetsBid hook we will be merging them.